File: src/main/java/com/chidozie/core/gaussianBlur/VerticalBlurShader.java
package com.chidozie.core.gaussianBlur;

import com.chidozie.core.shaders.ShaderProgram;

public class VerticalBlurShader extends ShaderProgram {

	private static final String VERTEX_FILE = "gaussianBlur/verticalBlurVertexShader.vert";
	private static final String FRAGMENT_FILE = "gaussianBlur/blurFragmentShader.frag";
	
	private int location_targetHeight;
	
	protected VerticalBlurShader() {
		super(VERTEX_FILE, FRAGMENT_FILE);
	}
	
	protected void loadTargetHeight(float height){
		super.loadFloat(location_targetHeight, height);
	}

	@Override
	protected void getAllUniformLocations() {	
		location_targetHeight = super.getUniformLocation("targetHeight");
	}

	@Override
	protected void bindAttributes() {
		super.bindAttribute(0, "position");
	}
}

File: src/main/kotlin/com/chidozie/core/gaussianBlur/VerticalBlurShader.kt
package com.chidozie.core.gaussianBlur

import com.chidozie.core.shaders.ShaderProgram

class VerticalBlurShader : ShaderProgram(VERTEX_FILE, FRAGMENT_FILE) {
    private var location_targetHeight = 0

    fun loadTargetHeight(height: Float) {
        super.loadFloat(location_targetHeight, height)
    }

    override fun getAllUniformLocations() {
        location_targetHeight = super.getUniformLocation("targetHeight")
    }

    override fun bindAttributes() {
        super.bindAttribute(0, "position")
    }

    companion object {
        private const val VERTEX_FILE = "gaussianBlur/verticalBlurVertexShader.vert"
        private const val FRAGMENT_FILE = "gaussianBlur/blurFragmentShader.frag"
    }
}
